src/components/InstanceFilters/InstanceFilters.js

```diff
@@ -99,23 +217,315 @@ const InstanceFilters = props => {
+    effectiveLocationOptions,
+    langOptions,
+    resourceTypeOptions,
+    instanceFormatOptions,
+    modeOfIssuanceOptions,
+    natureOfContentOptions,
+    suppressedOptions,
+    sourceOptions,
+    tagsRecords,
+  });
+
+  const accordionsData = {
+    effectiveLocation: { value: locationInputValue, isSelected: isLocationSelected },
+    language: { value: languageInputValue, isSelected: isLanguageSelected },
+    resource: { value: resourceInputValue, isSelected: isResourceSelected },
+    format: { value: formatInputValue, isSelected: isFormatSelected },
+    mode: { value: modeInputValue, isSelected: isModeSelected },
+    natureOfContent: { value: natureOfContentInputValue, isSelected: isNatureOfContentSelected },
+    staffSuppress: { value: staffSuppressInputValue, isSelected: isStaffSuppressSelected },
+    discoverySuppress: { value: discoverySuppressInputValue, isSelected: isDiscoverySuppressSelected },
+    source: { value: sourceInputValue, isSelected: isSourceSelected },
+    tags: { value: tagsInputValue, isSelected: isTagsSelected },
+  };
+
+  const handleFetchFacets = ({ onMoreClickedFacet, focusedFacet } = {}) => {
+    setTimeout(() => {
+      if (isAlreadyFetched.current) {
+        isAlreadyFetched.current = false;
+        return;
+      }
+
+      onFetchFacets({ onMoreClickedFacet, focusedFacet, accordions, accordionsData });
+    }, 300);
+  };
+
+  useEffect(() => {
+    const recordsNumber = Object.keys(records);
+    const facetsNumber = facetsOptions.length;
+
+    if (recordsNumber >= facetsNumber) {
+      setFacetsOptions(records);
+    } else {
+      setFacetsOptions(curFacetOptions => ({ ...curFacetOptions, ...records }));
+    }
+  }, [records])
+
+  useEffect(() => {
+    let facetOpenedFirstTime = '';
+
+    const isFacetOpenedFirstTime = _.some(prevAccordionsState.current, (prevFacetValue, facetName) => {
+      const curFacetValue = accordions[facetName];
+
+      if (curFacetValue !== prevFacetValue) {
+        if (curFacetValue) {
+          facetOpenedFirstTime = facetName;
+        }
+        return curFacetValue;
+      }
+      return false;
+    });
+
+    if (isFacetOpenedFirstTime) {
+      prevAccordionsState.current = accordions;
+      onFetchFacets({ facetOpenedFirstTime });
+      isAlreadyFetched.current = true;
+    }
+  }, [
+    accordions.effectiveLocation,
+    accordions.language,
+    accordions.resource,
+    accordions.format,
+    accordions.mode,
+    accordions.natureOfContent,
+    accordions.staffSuppress,
+    accordions.discoverySuppress,
+    accordions.createdDate,
+    accordions.updatedDate,
+    accordions.source,
+    accordions.tags,
+  ]);
+
+  useEffect(() => {
+    let isSomeFilterChanged = false;
+
+    if (effectiveLocation) {
+      const isLocationChanged = prevLocations.current.size !== effectiveLocation.length;
+      if (isLocationChanged) {
+        setIsLocationSelected(true);
+        accordionsData.effectiveLocation.isSelected = true;
```

Comment:
Ic belive it would be set anyway as setIsLocationSelected(true); would be foulfilled

Reply:
Thanks, fixed.